From b392f7d6ec7751e9aea40f0adf3fa022294d6979 Mon Sep 17 00:00:00 2001
From: Gerard Andre <gerard.andre@intel.com>
Date: Wed, 20 Apr 2016 10:55:16 +0200
Subject: [PATCH] [CHT-CR-MRD][Camera] Fix ov2680 AE oscillation issue.

The group holder isn't needed in CHT.

Change-Id: I4dd8525e8c3add86b0301a41b19a79cbd527d3bd
Signed-off-by: Lu, Xingjiang <xingjiangx.lu@intel.com>
Signed-off-by: Marco A Salgado Martinez <marco.antonio.salgado.martinez@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-28799
Depends on: Iedba89a8d8bf72f3c7f55fb1b670457667a16358
Reviewed-on: https://android.intel.com:443/496759
---
 drivers/media/i2c/atomisp/ov2680.c | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/drivers/media/i2c/atomisp/ov2680.c b/drivers/media/i2c/atomisp/ov2680.c
index 6d3d33d..722c827 100644
--- a/drivers/media/i2c/atomisp/ov2680.c
+++ b/drivers/media/i2c/atomisp/ov2680.c
@@ -409,15 +409,6 @@ static long __ov2680_set_exposure(struct v4l2_subdev *sd, int coarse_itg,
 	hts = ov2680_res[dev->fmt_idx].pixels_per_line;
 	vts = ov2680_res[dev->fmt_idx].lines_per_frame;
 
-	/* group hold */
-	ret = ov2680_write_reg(client, OV2680_8BIT,
-                                       OV2680_GROUP_ACCESS, 0x00);
-	if (ret) {
-		dev_err(&client->dev, "%s: write %x error, aborted\n",
-			__func__, OV2680_GROUP_ACCESS);
-		return ret;
-	}
-
 	/* Increase the VTS to match exposure + MARGIN */
 	if (coarse_itg > vts - OV2680_INTEGRATION_TIME_MARGIN)
 		vts = (u16) coarse_itg + OV2680_INTEGRATION_TIME_MARGIN;
@@ -491,17 +482,6 @@ static long __ov2680_set_exposure(struct v4l2_subdev *sd, int coarse_itg,
 		}
 	}
 
-	/* End group */
-	ret = ov2680_write_reg(client, OV2680_8BIT,
-			       OV2680_GROUP_ACCESS, 0x10);
-	if (ret)
-		return ret;
-
-	/* Delay launch group */
-	ret = ov2680_write_reg(client, OV2680_8BIT,
-					   OV2680_GROUP_ACCESS, 0xa0);
-	if (ret)
-		return ret;
 	return ret;
 }
 
-- 
1.9.1

